azurelinuxagent/common/future.py

```diff
@@ -61,14 +61,23 @@
     range = xrange
     int = long
 
-
     if sys.version_info[1] >= 7:
         from collections import OrderedDict  # For Py 2.7+
     else:
         from ordereddict import OrderedDict  # Works only on 2.6  # pylint: disable=E0401
 else:
     raise ImportError("Unknown python version: {0}".format(sys.version_info))
 
+#
+# distutils has been removed from Python >= 3.12; use the copy from azurelinuxagent instead
```

Comment:
if it's same copy, why don't we use it for all py versions? or eventually we remove distutils dependency when things work with copy?

Reply:
version.py is not identical across python versions; mostly syntactic differences, but i do not want to take the risk of introducing any changes in behavior across existing distros/python versions, so I am using the copy only on distros/python where there isn't a version.py

Reply:
(LooseVersion is used in the osutil hierarchy in many places, and that code is poorly tested)